swift/ios-shoppe-demo/ServiceManagers/APIService.swift

```diff
@@ -33,14 +33,32 @@ class APIService {
                 let json = try JSONSerialization.jsonObject(with: data, options: .mutableContainers) as? [Any]
 
                 for dict in json ?? [""] {
-                    self.products.append(Product(dict as? [String : Any] ?? [:]))
+                    products.append(Product(dict as? [String : Any] ?? [:]))
                 }
-                completion()
+                completion(products)
             }
             catch {
                 print("\(error.localizedDescription)") // TODO: Add proper error handling
-                completion()
+                completion(nil)
             }
         }.resume()
     }
+
+    func getProductsFromFile(completion: (_ :[Product])->()) {
+        var products = [Product]()
+
+        if let path = Bundle.main.path(forResource: "ShoppeItems", ofType: "js") {
```

Comment:
once the products file is updated to `.json` extensions, I imagine the `ofType` property will change.

Reply:
100%